test/api/navigation.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+const http = require("http");
+const zlib = require("zlib");
+const { assert } = require("chai");
+const { describe, it } = require("mocha-sugar-free");
+
+const { delay } = require("../util");
+const jsdom = require("../..");
+const { JSDOM } = require("../..");
+const { version: packageVersion } = require("../../package.json");
+
+require("chai").use(require("../chai-helpers.js"));
+
+describe("API: JSDOM.fromURL()", { skipIfBrowser: true }, () => {
+  it("should support navigation via location.replace", () => {
+    const [url1, url2] = twoPageServer("<p>Page 1</p>", "<p>Page 2</p>", 3);
+
+    return JSDOM.fromURL(url1).then(dom => {
+      assert.strictEqual(dom.serialize(), "<html><head></head><body><p>Page 1</p></body></html>");
+      dom.window.location.assign(url2);
+      return delay(200).then(() => {
```

Comment:
This delay makes me a bit nervous. I guess in the future there will be a JSDOM.on("navigate") or similar that can replace it?